src/metabase/api/automagic_dashboards.clj

```diff
@@ -35,68 +39,84 @@
+  [id show]
+  (-> id Segment api/check-404 (magic/automagic-analysis {:show (keyword show)})))
 
 (api/defendpoint GET "/segment/:id/:prefix/:rule"
   "Return an automagic dashboard analyzing segment with id `id`. using rule `rule`."
-  [id prefix rule]
+  [id prefix rule show]
   (-> id
       Segment
       api/check-404
-      (magic/automagic-dashboard (rules/load-rule (str prefix "/" rule ".yaml")))))
+      (magic/automagic-analysis
+       {:rule (load-rule "table" prefix rule)
+        :show (keyword show)})))
 
 (api/defendpoint GET "/question/:id/cell/:cell-query"
   "Return an automagic dashboard analyzing cell in question  with id `id` defined by
    query `cell-querry`."
-  [id cell-query]
+  [id cell-query show]
   (-> (query/adhoc-query {:query {:filter (decode-base64-json cell-query)}})
-      (magic/inject-segment (-> id Card api/check-404))
-      magic/automagic-dashboard))
+      (magic.comparison/inject-segment (-> id Card api/check-404))
+      (magic/automagic-analysis {:show (keyword show)})))
 
 (api/defendpoint GET "/question/:id/cell/:cell-query/:prefix/:rule"
   "Return an automagic dashboard analyzing cell in question  with id `id` defined by
    query `cell-querry` using rule `rule`."
-  [id cell-query prefix rule]
+  [id cell-query prefix rule show]
   (-> (query/adhoc-query {:query {:filter (decode-base64-json cell-query)}})
-      (magic/inject-segment (-> id Card api/check-404))
-      (magic/automagic-dashboard (rules/load-rule (str prefix "/" rule ".yaml")))))
+      (magic.comparison/inject-segment (-> id Card api/check-404))
+      (magic/automagic-analysis
+       {:rule (load-rule "table" prefix rule)
+        :show (keyword show)})))
 
 (api/defendpoint GET "/metric/:id"
   "Return an automagic dashboard analyzing metric with id `id`."
-  [id]
-  (-> id Metric api/check-404 magic/automagic-analysis))
+  [id show]
+  (-> id Metric api/check-404 (magic/automagic-analysis {:show (keyword show)})))
 
 (api/defendpoint GET "/field/:id"
   "Return an automagic dashboard analyzing field with id `id`."
-  [id]
-  (-> id Field api/check-404 magic/automagic-analysis))
+  [id show]
+  (-> id Field api/check-404 (magic/automagic-analysis {:show (keyword show)})))
 
 (api/defendpoint GET "/question/:id"
   "Return an automagic dashboard analyzing question with id `id`."
-  [id]
-  (-> id Card api/check-404 magic/automagic-analysis))
+  [id show]
+  (-> id Card api/check-404 (magic/automagic-analysis {:show (keyword show)})))
 
 (api/defendpoint GET "/adhoc/:query"
   "Return an automagic dashboard analyzing ad hoc query."
-  [query]
+  [query show]
   (-> query
       decode-base64-json
       query/adhoc-query
-      magic/automagic-analysis))
+      (magic/automagic-analysis {:show (keyword show)})))
+
+(api/defendpoint GET "/adhoc/:query/cell/:cell-query"
+  "Return an automagic dashboard analyzing ad hoc query."
+  [query cell-query show]
+  (let [query      (decode-base64-json query)
```

Comment:
What's going on here?

Reply:
It gets passed in two query dicts encoded with base64. `query` is the `:dataset_query` for the card, `cell-query` is a refinement `:filter` clause that defines a cell (eg. Country = Finland)